dataclasses.py

```diff
@@ -20,6 +21,16 @@ class FrozenInstanceError(AttributeError): pass
 # A sentinel object to detect if a parameter is supplied or not.
 _MISSING = object()
 
+# A sentinel object to signal that a default-factory will be used.
```

Comment:
Could you add a comment about why this class exists? It took me a while to understand it.

Also, why does it need `__eq__` and `__hash__`?

Reply:
I added a comment as suggested.

Indeed `__eq__` and `__hash__` are no longer needed (they were at one point).